sdk/core/Azure.Core.Experimental/tests/DynamicJsonTests.cs

```diff
@@ -278,6 +278,47 @@ public void CanMakeChangesAndAddNewProperty()
             Assert.AreEqual(2, (int)jsonData.Bar);
         }
 
+        [Test]
+        public void CanCheckOptionalProperty()
+        {
+            dynamic json = GetDynamicJson(@"
+                {
+                  ""Foo"" : ""foo""
+                }");
+
+            // Property is present
+            Assert.IsFalse((string)json.Foo == null);
+
+            // Property is absent
+            Assert.IsTrue((string)json.Bar == null);
+        }
+
+        [Test]
+        public void CanCheckOptionalPropertyWithChanges()
+        {
+            dynamic json = GetDynamicJson(@"
+                {
+                  ""Foo"" : ""foo"",
+                  ""Bar"" : {
+                    ""A"" : ""a""
+                  }
+                }");
+
+            // Add property Baz
+            json.Baz = "baz";
+
+            // Remove property A
+            json.Bar = new { B = "b" };
+
+            // Properties are present
+            Assert.IsFalse((string)json.Foo == null);
+            Assert.IsFalse((string)json.Bar.B == null);
+            Assert.IsFalse((string)json.Baz == null);
```

Comment:
What happens if we access a property on a non-existent property? e.g. `json.Baz.Fizz`?

Reply:
If you called `json.Baz.Fizz` on this example, it would throw because `json.Baz` is a string type in the JSON, and not an object that can have properties.

Reply:
What would the exception be?

Reply:
@annelo-msft, why is the cast to string required? Can we overload == on DynamicJson to avoid it?

Reply:
> @annelo-msft, why is the cast to string required? Can we overload == on DynamicJson to avoid it?

Yes, apologies that got lost across versions.  I'll add it back.

Reply:
> What would the exception be?

`System.InvalidOperationException: 'Expected an 'Object' type but was String.'`

Reply:
Do we have/need tests for these invalid property access scenarios?

Reply:
Not yet, I'll add them.  Thanks!

Reply:
> Why is the cast to string required? Can we overload == on DynamicJson to avoid it?

@KrzysztofCwalina, with overloads for `==`, when comparing a dynamic to `null`, we don't know the type of either, so you get an ambiguous call site, e.g.:

```
    Microsoft.CSharp.RuntimeBinder.RuntimeBinderException : The call is ambiguous between the following methods or properties: 'Azure.Core.Dynamic.DynamicJson.operator ==(Azure.Core.Dynamic.DynamicJson, string)' and 'Azure.Core.Dynamic.DynamicJson.operator ==(string, Azure.Core.Dynamic.DynamicJson)'
```

Without the overload and without the cast, it compares null to the returned DynamicJson object, which has a value so it's not null.  

One option is to return an actual `null` from GetProperty() if the property is missing.  We would lose the ability to treat is as a DynamicJson, but I'm not sure that's a problem.

Reply:
> One option is to return an actual null from GetProperty() if the property is missing. We would lose the ability to treat it as a DynamicJson, but I'm not sure that's a problem.

This is an interesting idea to try

Reply:
I've added follow-up items to track both these requests to https://github.com/Azure/azure-sdk-for-net/issues/33856